### DISCUSSION TITLE
Using SingleTaskMultiFidelityGP to make predictions on test data

### DISCUSSION THREAD

**Discussion by Abhijnan03** [meta-pytorch/botorch#1443] [Q&A] [answered]
I am trying to use the SingleTaskMultiFidelityGP model in BoTorch to train on high fidelity and low fidelity data and predict on a set of test data. The function I am trying to predict has only 1 dimension. However, when I try to compute the posterior of the model for the test data, I get an error stating "RuntimeError: Sizes of tensors must match except in dimension 0. Expected size 2 but got size 1 for tensor number 1 in the list.". I am attaching the code that I am using to do this and running it will recreate the error. 

Does the test data have to be in a different format for making a prediction? How can I compute the posterior for the SingleTaskMultiFidelityGP model?

[botorch_multifidelity.zip](https://github.com/pytorch/botorch/files/9736458/botorch_multifidelity.zip)

**Comment by Balandat** [comment, marked as answer]
I haven't looked a the code (on mobile) but this sounds like you're not including the fidelity parameter when making the predictions? Note that the model will predict at different fidelities, if you are interested in the predictions at the "target fidelity" you have to pass that in as part of your test data.
